### DISCUSSION TITLE
Problem in translating phrases with python format strings

### DISCUSSION THREAD

**Discussion by miaow2** [netbox-community/netbox#14957] [General]
Hi all,
I am approving translation for russian language at Transifex service and found problem with phrases that contain python format string.

For example, two help texts in CustomLinkForm https://github.com/netbox-community/netbox/blob/develop/netbox/extras/forms/model_forms.py#L143
These texts have a `{example}` format string, and Transifex does not allow me to add translation without it.

What to do in such situations? I suggest deleting format strings and adding "\<code\>{{ object }}\</code\>"  directly to the help texts.

**Comment by jeremystretch** [comment]
Format string variables should be carried over to the translated text without modification; `{example}` is just a token which represents a value to be resolved dynamically. For example, the English string

```
Cannot adopt {model} {name} as it already belongs to a module
```

is translated to Spanish as 

```
No puede adoptar {model} {name} porque ya pertenece a un módulo
```

> I suggest deleting format strings and adding `<code>{{ object }}</code>` directly to the help texts.

This instance in particular is necessary because Transifex was tripping up on the embedded code. We moved it for a format variable to fix automatic translation.

**Reply by miaow2** [reply]
thanks for your replay!
